packages/resolve-dependencies/src/index.ts

```diff
@@ -266,12 +268,21 @@ export default async function (
   }
 }
 
-function verifyPatches (patchedDependencies: string[], appliedPatches: Set<string>) {
+function verifyPatches (patchedDependencies: string[], appliedPatches: Set<string>, allowNonAppliedPatches: boolean = false) {
   const nonAppliedPatches: string[] = patchedDependencies.filter((patchKey) => !appliedPatches.has(patchKey))
   if (nonAppliedPatches.length) {
-    throw new PnpmError('PATCH_NOT_APPLIED', `The following patches were not applied: ${nonAppliedPatches.join(', ')}`, {
-      hint: 'Either remove them from "patchedDependencies" or update them to match packages in your dependencies.',
-    })
+    const code = 'PATCH_NOT_APPLIED'
+    const message = `The following patches were not applied: ${nonAppliedPatches.join(', ')}`
+    if (allowNonAppliedPatches) {
+      logger.warn({
+        prefix: code,
```

Comment:
The prefix should be a directory. In this case you can use globalWarn instead of logger.warn which is exported from `@pnpm/logger`. It doesn't need a prefix.